realm-export.json

```diff
@@ -0,0 +1,5384 @@
rrayList();\n\ngroups.forEach(scanTenant);\ngroups.forEach(scanGroups);\n\nfunction scanTenant(group){\n    if ( isTenant(group)===true ) {\n        found = getTenant(group.getName());\n        if ( !found ) {\n            var groups = new ArrayList();\n            var roles = new ArrayList();\n            var tenant = new HashMap();\n            tenant.put(\"id\",group.getId());\n            tenant.put(\"name\",group.getName());\n            tenant.put(\"groups\",groups);\n            tenant.put(\"roles\",roles);\n            tenants.add(tenant);\n        }\n    } else if( group !== null && group.getParent() !== null ) {\n        scanTenant(group.getParent());\n    }\n}\n\nfunction scanGroups(group){\n    if ( isTenant(group)===false )  {\n        found = getTenant(getRoot(group).getName());\n        if ( found ) {\n            groups = found.get(\"groups\");\n            roles = found.get(\"roles\");\n            current = getGroup(groups, group.getName());\n            if (! current ){\n                var rep = ModelToRepresentation.toRepresentation(group, true);\n                //if (rep.getRealmRoles())\n                //    roles.add(rep.getRealmRoles());\n                if( keycloakSession.getContext().getClient()){\n                    var clientId = \"resource_server\";\n                    if(rep.getClientRoles().get(clientId)) {\n                      roles.addAll(rep.getClientRoles().get(clientId));\n                    }\n                } else {\n                    var clients = realm.getClients();\n                     for (i= 0; i<clients.size(); i++){\n                        item = clients.get(i);\n                        var clientId = item.getClientId();\n                        if(rep.getClientRoles().get(clientId)) {\n                          roles.addAll(rep.getClientRoles().get(clientId));\n                        }\n                    }\n                }\n            }\n        }\n    }\n    if( group.getParent() !== null ) {\n        scanGroups(group.getParent());\n    }\n}\n\nfunction getTenant(name){\n    for (i= 0; i<tenants.size(); i++){\n        item = tenants.get(i);\n        if (item.get(\"name\")===name)\n            return item;\n    }\n    return null;\n}\n\nfunction getGroup(groups, name){\n    for (i= 0; i<groups.size(); i++){\n        item = groups.get(i);\n        if (item.get(\"name\")===name)\n            return item;\n    }\n    return null;\n}\n\nfunction getRoot(group){\n    if(group.getParent() !== null)\n        return getRoot(group.getParent());\n    return group;\n}\n\n\nfunction isServicePath(group){\n    if (group.getFirstAttribute(\"servicePath\") == \"true\" || group.getFirstAttribute(\"servicepath\") == \"true\")\n        return true;\n    return false;\n}\n\nfunction isTenant(group){\n    if (group.getParent()===null &&\n        group.getFirstAttribute(\"tenant\") == \"true\"){\n        return true;\n    }\n    return false;\n}\n\n\n\ntoken.setOtherClaims(\"tenant_roles\", tenants);",
```

Comment:
@Cerfoglg new versions of keycloak do not allow importing scripts in these way.
to this aim we have created the repo here:
https://github.com/orchestracities/keycloak-scripts

I think i already mentioned the repo and the fact that we should update scripts there as needed by the new auth mechanism